gateway/envs.go

```diff
@@ -52,3 +51,30 @@ func (g *Gateway) GetEnvsWithProjectToken(ctx context.Context) (*entity.Envs, er
 
 	return resp.Envs, nil
 }
+
+func (g *Gateway) UpdateEnvsForPlugin(ctx context.Context, req *entity.UpdateEnvsRequest) (*entity.Envs, error) {
+	gqlReq := gql.NewRequest(`
+	  	mutation($projectId: String!, $environmentId: String! $pluginId: String! $envs: Json!) {
+			updateEnvsForPlugin(projectId: $projectId, environmentId: $environmentId, pluginId: $pluginId, envs: $envs)
+	  	}
+	`)
+
+	err := g.authorize(ctx, gqlReq.Header)
+	if err != nil {
+		return nil, err
+	}
+
+	gqlReq.Var("projectId", req.ProjectID)
+	gqlReq.Var("environmentId", req.EnvironmentID)
+	gqlReq.Var("pluginId", req.PluginID)
+	gqlReq.Var("envs", req.Envs)
+
+	var resp struct {
+		Envs *entity.Envs `json:"updateEnvsForPlugin"`
+	}
+	if err := g.gqlClient.Run(ctx, gqlReq, &resp); err != nil {
+		return nil, err
+	}
+
+	return &entity.Envs{}, nil
```

Comment:
should this return resp.envs not an empty entity?

Reply:
Oh, nice catch! Had it like that because it was failing to deserialize without https://github.com/railwayapp/mono/pull/905